azure-pipelines.yml

```diff
@@ -0,0 +1,55 @@
+
+
+trigger:
+- master
+- develop
+- ft*
+
+variables:
+  ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+    appName: "pr$(System.PullRequest.PullRequestNumber)"
+  ${{ if not(eq(variables['Build.Reason'], 'PullRequest')) }}:
+    appName: "$(Build.SourceBranchName)"
+  azureSubscription: question-queue-pipeline
+  functionappName: queuefunction$(appName)
+  resourceGroup: math-dojo-hzprod-question-queue-function
+pool:
+  vmImage: 'windows-2019'
+
+steps:
+- task: AzureCLI@2
+  displayName: 'Create AppService'
+  inputs:
+    azureSubscription: question-queue-pipeline
+    scriptType: ps
+    scriptLocation: inlineScript
+    inlineScript: |
+       $serviceNameExists = az storage account check-name --name queuefunction$(appName) --subscription 5594cd6c-b674-4323-8517-5e859a399468 | ConvertFrom-Json
+        Write-Host "Does the service name exist for queuefunction$(appName): $serviceNameExists"
+        If ($serviceNameExists.nameAvailable) {
+            az storage account create --name queuefunction$(appName) --kind StorageV2 --location uksouth --resource-group $(resourceGroup) --subscription 5594cd6c-b674-4323-8517-5e859a399468
+            az functionapp create   --resource-group $(resourceGroup) --consumption-plan-location uksouth --name queuefunction$(appName) --storage-account  queuefunction$(appName) --os-type Windows --runtime node --functions-version 3 --disable-app-insights true
```

Comment:
Comparing this and the question/user account service, all functions in those get their own app service plans. In this case, we put all our resources in the same plan (we haven't specified via `--plan` so azure assumes we want to). Are there any benefits from sharing the same plan?